proxmox/virtual_environment_nodes.go

```diff
@@ -246,12 +252,42 @@ func (c *VirtualEnvironmentClient) OpenNodeShell(
 	})
 
 	sshConfig := &ssh.ClientConfig{
-		User:              ur[0],
+		User:              sshuser,
 		Auth:              []ssh.AuthMethod{ssh.Password(c.Password)},
 		HostKeyCallback:   cb,
 		HostKeyAlgorithms: kh.HostKeyAlgorithms(sshHost),
 	}
 
+	tflog.Info(ctx, fmt.Sprintf("Agent is set to %t", c.Agent))
+
+	if c.Agent {
```

Comment:
We may also want consider using password auth as a fallback in agent doesn't work for some reason

Reply:
I have implemented the fallback logic by making the ssh-agent branching a function that can return an error or a valid client.